bookworm/annotation/__init__.py

```diff
@@ -226,9 +223,17 @@ def onCaretMoved(self, event):
                 evtdata["line_contains_highlight"] = True
                 break
         for comment in NoteTaker(self.reader).get_for_page():
-            if comment.position in pos_range:
-                evtdata["comment"] = True
-                break
+            start_pos, end_pos = (comment.start_pos, comment.end_pos)
+            # If the comment has a selection, we check if the caret position is inside the selection. Otherwise, we check that the ocmment position is in the pos_range, typically the whole line.
+            condition = (start_pos <= position <= end_pos) if (start_pos, end_pos) != (None, None) else  comment.position in pos_range
```

Comment:
I noticed that we're using  <= end_pos  in this condition. I would suggest switching this to  < end_pos  instead. Using  <= end_pos  could cause an issue where the character immediately after  end_pos  is also considered part of the comment range.
```suggestion
            condition = (start_pos <= position < end_pos) if (start_pos, end_pos) != (None, None) else  comment.position in pos_range
```

Reply:
Hello. Sorry, I'm confused, how could it do so? end_pos refers to the end position of the comment. using <= simply includes the end position as well rather than excluding it.

Reply:
Suppose I add a comment to "world" in "Hello world!". When I review comment, Bookworm will treat the '!' as part of the comment.
In other words, our judgment condition here causes the comment range to be one more character than the real comment range.

Reply:
Oh yeah. Good catch.